app/src/context/slices/projectSlice.ts

```diff
@@ -0,0 +1,1986 @@
+        );
+      }
+
+      // fetch the existing remote DB
+      const remoteDb = databaseService.getRemoteDatabase(
+        project.database.remote.remoteDbId
+      );
+      if (!remoteDb) {
+        throw new Error(
+          `The remote DB with ID ${project.database.remote.remoteDbId} does not exist, so cannot update connection.`
+        );
+      }
+
+      // cleanup old sync
+      let updatedSyncId: undefined | string = undefined;
+
+      // Remove if needed
+      const oldSyncId = project.database.remote.syncId;
+      if (project.database.isSyncing && oldSyncId) {
+        databaseService.closeAndRemoveSync(oldSyncId);
+
+        updatedSyncId = buildSyncId({
+          localId: project.database.localDbId,
+          remoteId: project.database.remote.remoteDbId,
+        });
+
+        const handlers = createSyncStateHandlers(
+          payload.projectId,
+          payload.serverId,
+          store.dispatch
+        );
+        // creates the sync object (PouchDB.Replication.Sync)
+        const sync = createPouchDbSync({
+          // START syncing attachments
+          attachmentDownload: true,
+          // local is fine to continue using!
+          localDb,
+          // reuse existing remote db
+          remoteDb,
+          eventHandlers: handlers,
+        });
+
+        // Register the sync
+        databaseService.registerSync(updatedSyncId, sync);
+      }
+
+      // updates the state with all of this new information
+      state.servers[payload.serverId].projects[payload.projectId] = {
+        // These are retained
+        metadata: project.metadata,
+        projectId: project.projectId,
+        rawUiSpecification: project.rawUiSpecification,
+        uiSpecificationId: project.uiSpecificationId,
+        createdAt: project.createdAt,
+        lastUpdated: project.lastUpdated,
+        serverId: project.serverId,
+
+        // These are updated
+        isActivated: true,
+        database: {
+          // retained
+          isSyncing: project.database.isSyncing,
+          // This is UPDATED
+          isSyncingAttachments: true,
+          // This is retained
+          localDbId: project.database.localDbId,
+          remote: {
+            // reuse connection and remote
+            connectionConfiguration:
+              project.database.remote.connectionConfiguration,
+            remoteDbId: project.database.remote.remoteDbId,
+            // new sync
+            syncId: updatedSyncId,
+          },
+        },
+      };
+    },
+
+    /**
+     * Updates the sync state for a specific project
+     */
+    setSyncState: (
+      state,
+      action: PayloadAction<
+        ProjectIdentity & {
+          syncState: Partial<SyncState>;
+        }
+      >
+    ) => {
+      const {projectId, serverId, syncState} = action.payload;
+
+      // Check the server exists
+      const server = serverById(state, serverId);
+      if (!server) {
```

Comment:
I'm seeing a crash here where localDBId is not set, localDB has a value though and it looks like a db id.

Reply:
Refactor causes this issue due to you having the old version of the store. As discussed.